solr/core/src/java/org/apache/solr/util/SolrCLI.java

```diff
@@ -1316,19 +1076,20 @@ protected void runCloudTool(CloudLegacySolrClient cloudSolrClient, CommandLine c
             q = new SolrQuery("*:*");
             q.setRows(0);
             q.set(DISTRIB, "false");
-            try (HttpSolrClient solr = new HttpSolrClient.Builder(coreUrl).build()) {
-
-              String solrUrl = solr.getBaseURL();
-
-              qr = solr.query(q);
+            int lastSlash = coreUrl.substring(0, coreUrl.length() - 1).lastIndexOf('/');
+            try (var solrClient = getSolrClient(coreUrl.substring(0, lastSlash))) {
+              qr = solrClient.query(coreUrl.substring(lastSlash + 1, coreUrl.length() - 1), q);
```

Comment:
I would strongly recommend declaring a variable "collectionName" so that the variable itself can document *what* you are parsing

Reply:
I see more what you are doing.  You don't actually have to do any of this parsing.  Just call getSolrClient(coreUrl) (no substring stuff), and then when you call solrClient.query, call the overlaoded version without the collection / core name.